src/software/ai/navigator/path_planner/hrvo/hrvo_simulator.cpp

```diff
@@ -0,0 +1,242 @@
+#include "hrvo_simulator.h"
+
+HRVOSimulator::HRVOSimulator() : robots(), world(std::nullopt), primitive_set() {}
+
+void HRVOSimulator::updateWorld(const World &world,
+                                const RobotConstants_t &robot_constants,
+                                Duration time_step)
+{
+    this->world = world;
+    // TODO (#2498): Update implementation to correctly support adding and removing agents
+    //               to represent the newly added and removed friendly/enemy robots in the
+    //               World.
```

Comment:
This updated implementation technically supports adding/removing agents, so I think we can remove this. This implementation will need to be updated if we end up having some state that we need to keep from old Agents, but for now I think it should be okay

The one thing that will need to be updated is the `updateVelocity` call in Thunderloop:
https://github.com/UBC-Thunderbots/Software/blob/7760d348f9cf31348d98af7b27f1cad15d5b7cf9/src/software/jetson_nano/thunderloop.cpp#L205-L207
Since for this velocity feedback to actually be taken into account, it will need to be after the `updateWorld` call (since if it's before it, it will get overwritten with the World velocity), and it will need to be before the `stepPrimitive` call so HRVO actually uses the velocity. 
So I think you will probably need to cache the velocity after `MotorService::poll` for the next iteration of Thunderloop (might want to make the value `optional` so the value isn't used for the first iteration)